cli/template.go

```diff
@@ -272,25 +274,62 @@ func doTemplate(c *cli.Context) {
 	}
 }
 
-func calculateAllowedProperties(schemas []*SchemaWithPolicy, role string, manager *schema.Manager) {
-	// TODO: API document for domain scoped
-	auth := schema.NewAuthorizationBuilder().
-		WithRoleIDs(role).
-		WithKeystoneV2Compatibility().
-		BuildScopedToTenant()
+func calculateAllowedProperties(schemas []*SchemaWithPolicy, scopes []schema.Scope, role string, manager *schema.Manager, keystoneV2Compatible bool) {
+	if keystoneV2Compatible {
+		auth := schema.NewAuthorizationBuilder().
+			WithRoleIDs(role).
+			WithKeystoneV2Compatibility().
+			BuildScopedToTenant()
 
+		calculateAllowedPropertiesForAuth(schemas, auth, manager)
+	} else {
```

Comment:
Nit: I'd prefer to return in the true branch, so that the else branch and on level of indentation are not needed.

Reply:
done